modules/ding_provider/connie/connie.auth.inc

```diff
@@ -10,30 +10,20 @@
 function connie_auth_single_sign_on($name) {
   $res = array();
 
-  // Default password is the four last letters in the name.
-  $password = drupal_substr($name, -4);
   $passes = variable_get('connie_user_passes', array());
-  if (isset($passes[$name])) {
-    $password = $passes[$name];
-  }
-  
+
   // We simply check that the name existed in the $passes, which is basically
   // what we do at single sign on as log in is validated external.
-  if ($password) {
+  if (isset($passes[$name])) {
     $res['success'] = TRUE;
-    if (preg_match('/blocked/', $name)) {
-      $res['success'] = FALSE;
-      $res['messages'][] = t("Sorry, you're blocked");
-    }
-
     $res['creds'] = array(
       'name' => $name,
     );
 
     $res['user'] = array(
-      'mail' => $name . '@example.com',
+      'mail' => hash('crc32', $name) . '@example.com',
       'data' => array(
-        'display_name' => drupal_ucfirst($name),
+        'display_name' => hash('crc32', $name),
```

Comment:
As said, I'd go with `preg_replace('(\d{6})(\d{4})', '\1XXXX', $name);` as remembering what hash values different CPRs map to isn't that user friendly. Alternatively `preg_replace('(\d{6})(\d{4})', '\1-' . hash('crc32', $name), $name);` to support users with the same birthday.

Reply:
I have used this approach in #48.